src/services/Hooks/useCheckPermissions.ts

```diff
@@ -0,0 +1,41 @@
+import * as Notifications from 'expo-notifications';
+import * as Permissions from 'expo-permissions';
+import axios from 'axios';
+import { secretIp } from '../../../secrets/secrets';
+import { AsyncStorage } from 'react-native';
+import { extractId } from '../../store/helper';
+
+export const useCheckPermissions = async () => {
+    let accesskey = await AsyncStorage.getItem('access_token');
+    const userId = extractId(accesskey);
+
+    try {
+        let pushToken;
+        let permissionStatus = await Permissions.getAsync(Permissions.NOTIFICATIONS);
+        if (permissionStatus.status !== 'granted') {
+            permissionStatus = await Permissions.askAsync(Permissions.NOTIFICATIONS);
+        }
+        if (permissionStatus.status !== 'granted') {
```

Comment:
`pushToken = permissionStatus.status === 'granted' && (await Notifications.getExpoPushTokenAsync()).data; `

Reply:
what does the && do here? also, this doesn't include the pushToken = null portion

Reply:
`a = (b ===c) && true` is the same as  `a = (b ===c) ? true : null`

Reply:
so yes it does include the `pushToken = null` portion

Reply:
ok im including in next PR which is coming right up!